### PR TITLE
Parse Ivy-compatible RawRepository resolvers

### PR DESCRIPTION
**Pull request by er1c**
This fixes my [s3 plugin](https://github.com/frugalmechanic/fm-sbt-s3-resolver) resolver issues.

image

### CODE REVIEW

File: modules/lm-coursier/src/main/scala/coursier/lmcoursier/FromSbt.scala [hunk 1]
```diff
@@ -259,43 +260,77 @@ object FromSbt {
...
+    Option(v).map{ _.asScala.toVector }.getOrElse(Vector.empty).collect {
+      case s: String => s
+    }
+  }
+
+  implicit def toPatterns(resolver: IBiblioResolver): Patterns = Patterns(
+    ivyPatterns = toStringVector(resolver.getIvyPatterns),
+    artifactPatterns = toStringVector(resolver.getArtifactPatterns),
+    isMavenCompatible = resolver.isM2compatible,
+    descriptorOptional = resolver.isUseMavenMetadata,
```
Comment (er1c) [inline modules/lm-coursier/src/main/scala/coursier/lmcoursier/FromSbt.scala]: This might need to be a `!`

**Comment by alexarchambault** [comment]
@er1c Thanks!

It would be nice to have a small scripted test for that. See for example the one under `modules/sbt-coursier/src/sbt-test/shared-2/maven-plugin-classpath-type`, that can be run via `sbt-coursier/scripted shared-2/maven-plugin-classpath-type` at the sbt prompt (or just `sbt "sbt-coursier/scripted shared-2/maven-plugin-classpath-type"` at the shell).

A new one can be created under `modules/sbt-coursier/src/sbt-test/shared-2/s3` say, with a `check` task like `maven-plugin-classpath-type`, that could get `coursierResolvers`, call `FromSbt.repository` on its values, and check that it returns a repository for each of them.

Would you feel like adding one?

**Comment by er1c** [comment]
Sure, i'll try and explore this later today/tonight

**Comment by er1c** [comment]
Test added

**Comment by alexarchambault** [comment]
Thanks @er1c! I pushed some nit changes, I'll merge as soon as the CI is green.

**Pull request merged**